packages/realm/src/errors.ts

```diff
@@ -138,3 +140,45 @@ export class ClientResetError extends SyncError {
     };
   }
 }
+
+/**
+ * An error class that indicates that one or more object changes have been reverted by the server.
+ * This can happen when the client creates/updates objects that do not match any subscription, or performs writes on
+ * an object it didn't have permission to access.
+ */
+export class CompensatingWriteError extends SyncError {
+  /**
+   * The array of compensating writes performed by the server.
+   */
+  public infos: CompensatingWriteInfo[] = [];
```

Comment:
I'd suggest using e.g. `info` or `writes`.
```suggestion
  public writes: CompensatingWriteInfo[] = [];
```